connectivity/check/policy.go

```diff
@@ -333,6 +359,12 @@ var (
 		Drop:     true,
 		ExitCode: ExitCurlHTTPError,
 	}
+
+	// ResultOKWithMetricForwardCountTotal expects OK and
+	// a metric of the number of total of forward count increases.
+	ResultOKWithMetricForwardCountTotal = Result{
```

Comment:
I think we should make this a method on `Result`, that also takes in the metric names as an argument. So in expectations, you could use something like the following in `suite.go`:

```go
WithExpectations(func(a *check.Action) (egress, ingress check.Result) {
	return check.ResultOK.ExpectMetricsIncrease("cilium_forward_count_total", "cilium_foo_bar_total"), check.ResultOk
})
```

This allows you to easily test for drops too:
```go
WithExpectations(func(a *check.Action) (egress, ingress check.Result) {
	return check.ResultDrop.ExpectMetricsIncrease("cilium_dropped_count_total"), check.ResultDrop
})
```

Reply:
I did some refactoring to isolate more Result and its associated methods, which might not be the best idea. It is in a new commit (the last one) for the moment but I will rework the history if we go with that.

Reply:
Fixed.